thrift/sql/serializing.go

```diff
@@ -0,0 +1,60 @@
+package sql
+
+import (
+	"database/sql/driver"
+
+	"github.com/upfluence/errors"
+	"github.com/upfluence/thrift/lib/go/thrift"
+
+	"github.com/upfluence/pkg/thrift/serializer"
+	"github.com/upfluence/pkg/thrift/thriftutil"
+)
+
+var (
+	binaryThriftSerializer = serializer.NewTSerializer(
+		thriftutil.BinaryProtocolFactory,
+	)
+
+	binaryThriftDeserializer = serializer.NewTDeserializer(
+		thriftutil.BinaryProtocolFactory,
+	)
+)
+
+type TStructPtr[T any] interface {
+	thrift.TStruct
+	*T
+}
+
+type NullThrift[T any, PT TStructPtr[T]] struct {
+	Data PT
```

Comment:
if you want to make it public can you add:
* `SerializerFactory *serializer.TSerializerFactory`
* a method w a fall back

```
func (nt NullThrift[T, PT]) serializerFactory() serializer.TSerializerFactory {
  if nt.SerializerFactory != nil {
    return nt.SerializerFactory
  }

  return defaultSerializerFactory
}
```

3- replace your global variable by a single `var defaultSerializerFactory = serializer.NewDefaultTSerializerFactory()`